### PR TITLE
Refactor externalmodule to es6

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

refactor

**Did you add tests for your changes?**

N/A

**If relevant, link to documentation update:**

N/A

**Summary**

refactor external module to esnext

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: lib/ExternalModule.js [hunk 1]
```diff
@@ -2,92 +2,115 @@
...
+		const objectLookup = moduleAndSpecifiers.slice(1).map(r => `[${JSON.stringify(r)}]`).join("");
+		return `module.exports = require(${moduleName})${objectLookup};`;
 	}
-	if(this.useSourceMap) {
-		return new OriginalSource(str, this.identifier());
-	} else {
-		return new RawSource(str);
+
+	checkExternalVariable(variableToCheck, request) {
+		return `if(typeof ${variableToCheck} === 'undefined') {${WebpackMissingModule.moduleCode(request)}}
```
Comment (sokra) [inline lib/ExternalModule.js]: `\n` instead of line feed

File: lib/ExternalModule.js [hunk 2]
```diff
@@ -2,92 +2,115 @@
...
+`;
 	}
-};
 
-ExternalModule.prototype.size = function() {
-	return 42;
-};
+	getSourceForAmdOrUmdExternal(id, optional, request) {
+		const externalVariable = `__WEBPACK_EXTERNAL_MODULE_${id}__`;
```
Comment (sokra) [inline lib/ExternalModule.js]: `` `__WEBPACK_EXTERNAL_MODULE_${id}__` `` -> ``Template.toIdentifier(`__WEBPACK_EXTERNAL_MODULE_${id}__`)``

just to be safe.

Comment (sokra) [inline lib/ExternalModule.js]: The same in the LibraryTemplateSomething where `__WEBPACK_EXTERNAL_MODULE_` is also used.

**Comment by timse** [comment]
interesting, the `Template.toIdentifier` change, may actually fix some existing bugs :)

**Comment by sokra** [comment]
Thanks

**Pull request merged**